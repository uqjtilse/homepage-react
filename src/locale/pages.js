/* eslint-disable max-len */
import React from 'react';
import { StandardCard } from 'modules/SharedComponents/Toolbox/StandardCard';
/*

NOTE:
- text can be either plain text, eg text: 'Some text to display' or
- text can be formatted HTML text, eg
    text: (<div>Click here to search google: <a href='google.com'>search google</a></div>)
IMPORTANT: if currently text contains placeholders, eg any characters in square brackets,
eg [noOfResults] it cannot be formatted with HTML tags’

- help objects have the following shape:
help: {
    title: 'About these metrics',
    text: (<div></div>),
    buttonLabel: 'CLOSE'
}
- text can be plain or formatted HTML component with links/tags/etc
- if help is not required, delete help: {} fully (including closing '},')

*/
/* eslint-disable max-len */
export default {
    pages: {
        index: {
            title: 'eSpace',
        },
        contact: {
            title: 'Contact UQ Library',
            children: (
                <StandardCard>
                    <h3>Awaiting content</h3>
                    <p>Content will go here.</p>
                </StandardCard>
            ),
        },
        notFound: {
            title: 'Page not found',
            children: (
                <StandardCard>
                    <p>The requested page could not be found.</p>
                    <p>Sorry about that, but here's what you can do next:</p>
                    <ul>
                        <li>Try re-typing the address, checking for spelling, capitalisation and/or punctuation.</li>
                        <li>Start again at the home page.</li>
                        <li>
                            If you’re sure the page should be at this address, email us at webmaster@library.uq.edu.au.
                        </li>
                    </ul>
                </StandardCard>
            ),
        },
        masquerade: {
            title: 'Masquerade',
            help: {
                title: 'Masquerade',
                text: <p>Masquerade as another user...</p>,
                buttonLabel: 'CLOSE',
            },
            description: {
                readonly: (
                    <p>
                        <strong>NOTE:</strong> As a read-only masquerader, you can view all parts of the profile, but
                        you are not able to make any changes to the account.
                    </p>
                ),
                full: (
                    <p>
                        <strong>WARNING!!</strong> When masquerading as a user, you will effectively become that user,
                        and changes you make will apply to the account!
                    </p>
                ),
            },
            labels: {
                submit: 'Masquerade',
                hint: 'Enter a UQ staff or student username (eg. uqjsmith1 or s123456)',
            },
        },
    },
};
