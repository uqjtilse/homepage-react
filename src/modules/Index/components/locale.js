import React from 'react';
import PublicIcon from '@material-ui/icons/Public';
import ImportContactsIcon from '@material-ui/icons/ImportContacts';
import DescriptionIcon from '@material-ui/icons/Description';
import MovieIcon from '@material-ui/icons/Movie';
import InboxIcon from '@material-ui/icons/Inbox';
import StorageIcon from '@material-ui/icons/Storage';
import FindInPageIcon from '@material-ui/icons/FindInPage';
import ListAltIcon from '@material-ui/icons/ListAlt';
import SchoolIcon from '@material-ui/icons/School';

export default {
    PrimoSearch: {
        typeSelect: {
            label: 'Search',
            items: [
                {
                    name: 'Library',
                    icon: <PublicIcon size="small" color="secondary" />,
                    link:
                        'https://search.library.uq.edu.au/primo-explore/search?query=any,contains,[keyword]&tab=61uq_all&search_scope=61UQ_All&sortby=rank&vid=61UQ&offset=0&facet=rtype,exclude,newspaper_articles,lk&facet=rtype,exclude,reviews,lk',
                },
                {
                    name: 'Books',
                    icon: <ImportContactsIcon size="small" color="secondary" />,
                    link:
                        'https://search.library.uq.edu.au/primo-explore/search?query=any,contains,[keyword]&tab=61uq_all&search_scope=61UQ_All&sortby=rank&vid=61UQ&offset=0&facet=rtype,include,books,lk',
                },
                {
                    name: 'Journal articles',
                    icon: <SchoolIcon size="small" color="secondary" />,
                    link:
                        'https://search.library.uq.edu.au/primo-explore/search?query=any,contains,[keyword]&tab=61uq_all&search_scope=61UQ_All&vid=61UQ&offset=0&fctV=articles&facet=rtype,include,articles,lk',
                },
                {
                    name: 'Video & audio',
                    icon: <MovieIcon size="small" color="secondary" />,
                    link:
                        'https://search.library.uq.edu.au/primo-explore/search?query=any,contains,[keyword]&tab=61uq_all&search_scope=61UQ_All&sortby=rank&vid=61UQ&offset=0&facet=rtype,include,media,lk',
                },
                {
                    name: 'Journals',
                    icon: <DescriptionIcon size="small" color="secondary" />,
                    link:
                        'https://search.library.uq.edu.au/primo-explore/search?query=title,contains,[keyword],AND&tab=61uq_all&vid=61UQ&search_scope=61UQ_All&sortby=rank&mfacet=rtype,include,journals,1,lk&mode=advanced&offset=0',
                },
                {
                    name: 'Physical items',
                    icon: <InboxIcon size="small" color="secondary" />,
                    link:
                        'https://search.library.uq.edu.au/primo-explore/search?query=any,contains,[keyword]&tab=61uq_all&search_scope=61UQ_All&vid=61UQ&offset=0&facet=tlevel,include,physical_items,lk',
                },
                {
                    name: 'Databases',
                    icon: <StorageIcon size="small" color="secondary" />,
                    link:
                        'https://search.library.uq.edu.au/primo-explore/dbsearch?query=any,contains,[keyword]&tab=jsearch_slot&vid=61UQ&offset=0&databases=any,[keyword]',
                },
                {
                    name: 'Past exam papers',
                    icon: <FindInPageIcon size="small" color="secondary" />,
                    link: 'https://www.library.uq.edu.au/exams/papers.php?stub=[keyword]',
                },
                {
                    name: 'Course reading lists',
                    icon: <ListAltIcon size="small" color="secondary" />,
                    link: 'https://uq.rl.talis.com/search.html?q=[keyword]',
                },
            ],
        },
        links: [
            {
                label: 'Search help',
                link: 'https://web.library.uq.edu.au/research-tools-techniques/uq-library-search',
            },
            {
                label: 'Advanced search',
                link: 'https://search.library.uq.edu.au/primo-explore/search?vid=61UQ&sortby=rank&mode=advanced',
            },
            {
                label: 'Database search',
                link: '#',
            },
            {
                label: 'Browse search',
                link: 'https://search.library.uq.edu.au/primo-explore/browse?vid=61UQ&sortby=rank',
            },
        ],
    },
};
