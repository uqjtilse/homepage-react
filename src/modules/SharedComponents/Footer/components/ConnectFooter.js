import React from 'react';
import PropTypes from 'prop-types';

import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
    connectFooter: {
        fontFamily: 'Roboto, "Helvetica Neue", Helvetica, Arial, sans-serif',
        fontSize: '14px',
        fontWeight: '300',
        lineHeight: '25px',
        margin: '0 auto',
        maxWidth: '1200px',
        position: 'relative',
        '& a': {
            color: '#333',
            textDecoration: 'none',
        },
        '& div': {
            [theme.breakpoints.down('sm')]: {
                textAlign: 'center',
            },
        },
    },
    navigation: {
        padding: '20px',
        '& ul': {
            margin: '0 0 0 20px',
            padding: 0,
        },
        '& li': {
            listStyle: 'none',
            margin: 0,
            padding: 0,
            [theme.breakpoints.down('sm')]: {
                display: 'inline-block',
            },
        },
    },
    separator: {
        display: 'none',
        [theme.breakpoints.down('sm')]: {
            display: 'inline-block',
        },
    },
    buttonColoredAccent: {
        display: 'flex', // needed to get the wiiiide button for Giving buttons
        margin: '10px auto 0 auto',
        maxWidth: '244px',
        width: '100%',
        '& a': {
            backgroundColor: '#0e62eb',
            border: '1px solid #0e62eb',
            borderRadius: '3px',
            boxShadow:
                '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2)',
            color: '#fff',
            maxWidth: '244px',
            outlineWidth: 0,
            padding: '0.7em',
            textAlign: 'center',
            transition: 'box-shadow 0.28s cubic-bezier(0.4, 0, 0.2, 1)',
            width: '100%',
            '&:hover': {
                backgroundColor: '#fff',
                color: '#0e62eb !important',
                fontWeight: 'bold',
                textDecoration: 'none',
            },
        },
    },
    buttonSocial: {
        textAlign: 'center',
        '& div': {
            backgroundColor: '#000',
            borderRadius: '4px',
            height: '33px',
            margin: '0 5px',
            padding: '8px 0 0',
            width: '33px',
        },
        '& a': {
            color: '#fff',
            display: 'block',
            margin: 0,
            textAlign: 'center',
            textDecoration: 'none',
            textTransform: 'uppercase',
        },
        '& img': {
            height: '20px',
            lineHeight: 0,
            width: '20px',
        },
        [theme.breakpoints.down('sm')]: {
            textAlign: 'center',
        },
    },
    internal: {
        bottom: '1rem',
        position: 'absolute',
        [theme.breakpoints.down('sm')]: {
            bottom: 'auto',
            marginTop: '1rem',
            position: 'relative',
        },
    },
    giving: {
        '& div': {
            [theme.breakpoints.down('sm')]: {
                margin: '5px auto',
            },
        },
    },
    contacts: {
        '& div': {
            '& div': {
                [theme.breakpoints.down('sm')]: {
                    margin: '0 auto',
                },
            },
        },
    },
});

export function ConnectFooter(props) {
    const { classes } = props;
    return (
        <Grid className={`${classes.connectFooter}`} container spacing={3}>
            <Grid item xs={12} md={4} className={`${classes.navigation}`}>
                {/* if we continue with this display, it should be driven by the json that creates the megamenu... */}
                <ul>
                    <li>
                        <a href="http://www.library.uq.edu.au">Library home</a>
                    </li>
                    <li className={`${classes.separator}`}>&nbsp;|&nbsp;</li>
                    <li>
                        <a href="https://web.library.uq.edu.au/library-services">Library services</a>
                    </li>
                    <li className={`${classes.separator}`}>&nbsp;|&nbsp;</li>
                    <li>
                        <a href="https://web.library.uq.edu.au/research-tools-techniques">
                            Research tools &amp; techniques
                        </a>
                    </li>
                    <li className={`${classes.separator}`}>&nbsp;|&nbsp;</li>
                    <li>
                        <a href="https://web.library.uq.edu.au/collections">Collections</a>
                    </li>
                    <li className={`${classes.separator}`}>&nbsp;|&nbsp;</li>
                    <li>
                        <a href="https://web.library.uq.edu.au/borrowing-requesting">Borrowing &amp; requesting</a>
                    </li>
                    <li className={`${classes.separator}`}>&nbsp;|&nbsp;</li>
                    <li>
                        <a href="https://web.library.uq.edu.au/locations-hours">Locations &amp; hours</a>
                    </li>
                    <li className={`${classes.separator}`}>&nbsp;|&nbsp;</li>
                    <li>
                        <a href="https://web.library.uq.edu.au/about-us">About</a>
                    </li>
                    <li className={`${classes.separator}`}>&nbsp;|&nbsp;</li>
                    <li>
                        <a href="https://web.library.uq.edu.au/contact-us">Contact us</a>
                    </li>
                </ul>
            </Grid>
            <Grid item xs={12} md={4} className={`${classes.contacts}`}>
                <Grid container>
                    <Grid item>
                        <Typography
                            style={{ fontSize: '1.2rem', color: '#000', marginBottom: '10px', fontWeight: 400 }}
                            variant={'h4'}
                        >
                            Connect with us
                        </Typography>
                    </Grid>
                </Grid>
                <Grid container className={`${classes.buttonSocial}`}>
                    <Grid aria-disabled="false" item role="button" style={{ paddingTop: '7px' }} tabIndex="-1" xs={2}>
                        <a
                            href="https://web.library.uq.edu.au/blog"
                            rel="noopener noreferrer"
                            target="_blank"
                            title="Library Blog"
                        >
                            Blog
                        </a>
                    </Grid>
                    <Grid item xs={2} tabIndex="-1" role="button" aria-disabled="false">
                        <a
                            href="https://twitter.com/UQ_Library"
                            rel="external noopener noreferrer"
                            target="_blank"
                            title="Library on Twitter"
                        >
                            <img
                                alt="Twitter icon"
                                aria-label="Library on Twitter"
                                src="//assets.library.uq.edu.au/reusable-components/resources/social-media-icons/twitter.png"
                                title="Twitter"
                            />
                        </a>
                    </Grid>
                    <Grid item xs={2} tabIndex="-1" role="button" aria-disabled="false">
                        <a
                            href="https://www.facebook.com/uniofqldlibrary"
                            rel="external noopener noreferrer"
                            target="_blank"
                            title="Library on Facebook"
                        >
                            <img
                                alt="Facebook icon"
                                aria-label="Library on Facebook"
                                src="//assets.library.uq.edu.au/reusable-components/resources/social-media-icons/facebook.png"
                                title="Facebook"
                            />
                        </a>
                    </Grid>
                    <Grid item xs={2} role="button" aria-disabled="false">
                        <a
                            href="https://www.instagram.com/uniofqldlibrary/"
                            rel="external noopener noreferrer"
                            tabIndex="-1"
                            target="_blank"
                            title="Library on Instagram"
                        >
                            <img
                                alt="Instagram icon"
                                aria-label="Library on Instagram"
                                src="//assets.library.uq.edu.au/reusable-components/resources/social-media-icons/instagram.png"
                                title="Instagram"
                            />
                        </a>
                    </Grid>
                    <Grid item xs={2} tabIndex="-1" role="button" aria-disabled="false">
                        <a
                            href="https://www.youtube.com/user/uqlibrary"
                            rel="external noopener noreferrer"
                            target="_blank"
                            title="Library on YouTube"
                        >
                            <img
                                alt="YouTube icon"
                                aria-label="Library on YouTube"
                                src="//assets.library.uq.edu.au/reusable-components/resources/social-media-icons/youtube.png"
                                title="YouTube"
                            />
                        </a>
                    </Grid>
                </Grid>
                <div className={`${classes.internal}`}>
                    <a href="https://support.my.uq.edu.au/app/library/feedback">Feedback</a>
                    &nbsp;|&nbsp;{' '}
                    <a href="https://web.library.uq.edu.au/about-us/participate-customer-research">
                        Help us improve
                    </a>{' '}
                    &nbsp;|&nbsp; <a href="https://web.library.uq.edu.au/sitemap">Site Map</a>
                </div>
            </Grid>
            <Grid item xs={12} md={4} className={`${classes.giving}`}>
                <Grid
                    aria-disabled="false"
                    aria-label="Join Friends of the Library"
                    className={`${classes.buttonColoredAccent}`}
                    id="joinFriend"
                    item
                    role="button"
                    tabIndex="-1"
                    xs={12}
                >
                    <a href="https://web.library.uq.edu.au/about-us/friends-library">Join Friends of the Library</a>
                </Grid>
                <Grid
                    aria-disabled="false"
                    aria-label="Give to the Library"
                    className={`${classes.buttonColoredAccent}`}
                    id="giveToLibrary"
                    item
                    role="button"
                    tabIndex="-1"
                    xs={12}
                >
                    <a href="https://www.uq.edu.au/giving/organisations/university-queensland-library">
                        Give to the Library
                    </a>
                </Grid>
            </Grid>
        </Grid>
    );
}

ConnectFooter.propTypes = {
    className: PropTypes.string,
    classes: PropTypes.object.isRequired,
};

ConnectFooter.defaultProps = {
    className: '',
    classes: {},
};

export default withStyles(styles, { withTheme: true })(ConnectFooter);
