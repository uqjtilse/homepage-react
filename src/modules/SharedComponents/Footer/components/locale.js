import React from 'react';
import FacebookIcon from '@material-ui/icons/Facebook';
import InstagramIcon from '@material-ui/icons/Instagram';
import TwitterIcon from '@material-ui/icons/Twitter';
import YouTubeIcon from '@material-ui/icons/YouTube';

export default {
    connectFooter: {
        buttonSocialLabel: 'Connect with us',
        buttonSocial: [
            {
                dataTestid: 'connect-blog-link',
                linklabel: 'Blog',
                linktitle: 'Library Blog',
                linkTo: 'https://web.library.uq.edu.au/blog',
            },
            {
                dataTestid: 'connect-twitter-link',
                linktitle: 'Library on Twitter',
                linkTo: 'https://twitter.com/UQ_Library',
                icon: <TwitterIcon />,
            },
            {
                dataTestid: 'connect-facebook-link',
                linktitle: 'Library on Facebook',
                linkTo: 'https://www.facebook.com/uniofqldlibrary',
                icon: <FacebookIcon />,
            },
            {
                dataTestid: 'connect-instagram-link',
                linktitle: 'Library on Instagram',
                linkTo: 'https://www.instagram.com/uniofqldlibrary/',
                icon: <InstagramIcon />,
            },
            {
                dataTestid: 'connect-youtube-link',
                linktitle: 'Library on YouTube',
                linkTo: 'https://www.youtube.com/user/uqlibrary',
                icon: <YouTubeIcon />,
            },
        ],
        internalLinks: [
            {
                dataTestid: 'connect-feedback-link',
                linklabel: 'Feedback',
                linkTo: 'https://support.my.uq.edu.au/app/library/feedback',
            },
            {
                dataTestid: 'connect-participate-link',
                linklabel: 'Help us improve',
                linkTo: 'https://web.library.uq.edu.au/about-us/participate-customer-research',
            },
            {
                dataTestid: 'connect-sitemap-link',
                linklabel: 'Site Map',
                linkTo: 'https://web.library.uq.edu.au/sitemap',
            },
        ],
        givingLinks: [
            {
                dataTestid: 'connect-friends-link',
                label: 'Join Friends of the Library',
                linkTo: 'https://web.library.uq.edu.au/about-us/friends-library',
            },
            {
                dataTestid: 'connect-give-link',
                label: 'Give to the Library',
                linkTo: 'https://www.uq.edu.au/giving/organisations/university-queensland-library',
            },
        ],
    },
    minimalFooter: {
        leftColumn: {
            line1: [
                {
                    type: 'text',
                    text: 'Â© The University of Queensland',
                },
            ],
            line2: [
                {
                    type: 'text',
                    text: 'Enquiries:',
                },
                {
                    type: 'link',
                    dataTestid: 'footer-enquiries-link',
                    linkLabel: '+61 7 3365 1111',
                    linktitle: 'UQ Enquiries phone number',
                    linkTo: 'tel:+61733651111',
                },
                {
                    type: 'divider',
                },
                {
                    type: 'link',
                    dataTestid: 'footer-contacts-link',
                    linkLabel: 'Contact directory',
                    linkTo: 'http://uq.edu.au/contacts',
                },
            ],
            line3: [
                {
                    type: 'abbr',
                    abbrDisplay: 'ABN',
                    abbrMouseoverText: 'Australian Business Number',
                },
                {
                    type: 'space',
                },
                {
                    type: 'text',
                    text: '63 942 912 684:',
                },
                {
                    type: 'divider',
                },
                {
                    type: 'abbr',
                    abbrDisplay: 'CRICOS',
                    abbrabbrMouseoverText: 'Commonwealth Register of Institutions and Courses for Overseas Students',
                },
                {
                    type: 'space',
                },
                {
                    type: 'link',
                    dataTestid: 'footer-cricos-link',
                    linkLabel: '00025B',
                    linktitle: 'Provider number',
                    linkTo: 'https://www.uq.edu.au/about/cricos-link',
                },
            ],
        },
        rightColumn: {
            line1: [
                {
                    type: 'header',
                    text: 'Emergency',
                },
            ],
            line2: [
                {
                    type: 'text',
                    text: 'Phone:',
                },
                {
                    type: 'link',
                    dataTestid: 'footer-emergency-link',
                    linkLabel: '3365 3333',
                    linktitle: 'UQ Emergency phone number',
                    linkTo: 'tel:+61733653333',
                },
            ],
        },
        bottomBlock: {
            line1: [
                {
                    type: 'link',
                    dataTestid: 'footer-terms-link',
                    linkLabel: 'Privacy & Terms of use',
                    linkTo: 'https://www.uq.edu.au/terms-of-use/',
                },
                {
                    type: 'divider',
                },
                {
                    type: 'link',
                    dataTestid: 'footer-feedback-link',
                    linkLabel: 'Feedback',
                    linkTo: 'https://support.my.uq.edu.au/app/library/feedback',
                },
            ],
        },
    },
};
