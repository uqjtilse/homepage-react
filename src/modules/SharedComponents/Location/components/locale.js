export const locale = {
    tooltip: 'Current preferred location is [currentLocation] - Click to change',
    noLocationSet: 'not set',
    menuTitle: 'Select a preferred location',
    locations: [
        {
            location: 'No preference',
            value: 'not set',
        },
        {
            location: 'St Lucia',
            value: 'St Lucia',
        },
        {
            location: 'Gatton',
            value: 'Gatton',
        },
        {
            location: 'Herston',
            value: 'Herston',
        },
    ],
};
