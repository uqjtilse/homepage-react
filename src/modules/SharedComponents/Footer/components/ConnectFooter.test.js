import { default as locale } from '../footerLocale';
import { default as menuLocale } from 'locale/menu';

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
});

import { ConnectFooter } from './ConnectFooter';

function setup(testProps = {}) {
    const props = {
        ...testProps,
        classes: {},
        history: testProps.history || { push: jest.fn() },
    };
    return getElement(ConnectFooter, props);
}

describe('Component Megamenu', () => {
    it('should render', () => {
        const wrapper = setup();
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('should have a valid footer locale', () => {
        // because the footer locale is meant to be maintained by the user, we check the file is valid
        expect(locale.connectFooter.buttonSocialHeader.length).not.toBe(0);
        locale.connectFooter.buttonSocial.forEach(menu => {
            expect(menu.dataTestid.length).not.toBe(0);
            expect(menu.linkTo.length).not.toBe(0);
            expect(menu.linkMouseOver.length).not.toBe(0);
            (!!menu.linklabel && expect(menu.linklabel.length).not.toBe(0)) ||
                (!!menu.icon && expect(menu.icon.length).not.toBe(0));
        });

        locale.connectFooter.internalLinks.forEach(menu => {
            expect(menu.dataTestid.length).not.toBe(0);
            expect(menu.linklabel.length).not.toBe(0);
            expect(menu.linkTo.length).not.toBe(0);
        });

        locale.connectFooter.givingLinks.forEach(menu => {
            expect(menu.dataTestid.length).not.toBe(0);
            expect(menu.label.length).not.toBe(0);
            expect(menu.linkTo.length).not.toBe(0);
        });
    });

    it('should have a valid menu locale', () => {
        // because the menu locale is meant to be maintained by the user, we check the file is valid
        expect(menuLocale.menuhome.dataTestid.length).not.toBe(0);
        expect(menuLocale.menuhome.linkTo.length).not.toBe(0);
        expect(menuLocale.menuhome.primaryText.length).not.toBe(0);

        menuLocale.publicmenu.forEach(menu => {
            expect(menu.dataTestid.length).not.toBe(0);
            expect(menu.linkTo.length).not.toBe(0);
            expect(menu.primaryText.length).not.toBe(0);
            expect(menu.primaryText.length).not.toBe(0);
        });
    });
});
