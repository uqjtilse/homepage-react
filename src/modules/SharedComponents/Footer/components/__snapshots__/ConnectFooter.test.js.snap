// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Megamenu should render 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="flex-start"
  container={true}
  data-testid="connect-footer"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={4}
    xs={12}
  >
    <ul>
      <li>
        <a
          data-testid="footermenu-homepage"
          href="http://www.library.uq.edu.au"
        >
          Library home
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-0"
      >
        <a
          data-testid="connect-services-link"
          href="https://web.library.uq.edu.au/library-services"
        >
          Library services
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-1"
      >
        <a
          data-testid="connect-research-link"
          href="https://web.library.uq.edu.au/research-tools-techniques"
        >
          Research tools & techniques
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-2"
      >
        <a
          data-testid="connect-collections-link"
          href="https://web.library.uq.edu.au/collections"
        >
          Collections
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-3"
      >
        <a
          data-testid="connect-borrowing-link"
          href="https://web.library.uq.edu.au/borrowing-requesting"
        >
          Borrowing & requesting
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-4"
      >
        <a
          data-testid="connect-locations-link"
          href="https://web.library.uq.edu.au/locations-hours"
        >
          Locations & hours
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-5"
      >
        <a
          data-testid="connect-about-link"
          href="https://web.library.uq.edu.au/about-us"
        >
          About
        </a>
        <Hidden
          mdUp={true}
        >
           | 
        </Hidden>
      </li>
      <li
        key="footerli-6"
      >
        <a
          data-testid="connect-contact-link"
          href="https://web.library.uq.edu.au/contact-us"
        >
          Contact us
        </a>
      </li>
    </ul>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={4}
    xs={12}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs="auto"
      >
        <WithStyles(ForwardRef(Typography))
          component="h3"
          variant="h6"
        >
          Connect with us
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={1}
    >
      <Hidden
        mdUp={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        />
      </Hidden>
      <WithStyles(ForwardRef(Grid))
        id="buttonSocial-0"
        item={true}
        key="buttonSocial-0"
        xs="auto"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionProps={
            Object {
              "timeout": 300,
            }
          }
          id="auth-button-0"
          placement="bottom"
          title="Visit the Library Blog"
        >
          <WithStyles(ForwardRef(Button))
            classes={
              Object {
                "root": undefined,
              }
            }
            color="primary"
            data-testid="socialbutton-dataTestid-0"
            id="socialbutton-0"
            onClick={[Function]}
            variant="contained"
          >
            Blog
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        id="buttonSocial-1"
        item={true}
        key="buttonSocial-1"
        xs="auto"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionProps={
            Object {
              "timeout": 300,
            }
          }
          id="auth-button-1"
          placement="bottom"
          title="Visit the Library on Twitter"
        >
          <WithStyles(ForwardRef(Button))
            classes={
              Object {
                "root": undefined,
              }
            }
            color="primary"
            data-testid="socialbutton-dataTestid-1"
            id="socialbutton-1"
            onClick={[Function]}
            variant="contained"
          >
            <TwitterIcon />
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        id="buttonSocial-2"
        item={true}
        key="buttonSocial-2"
        xs="auto"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionProps={
            Object {
              "timeout": 300,
            }
          }
          id="auth-button-2"
          placement="bottom"
          title="Visit the Library on Facebook"
        >
          <WithStyles(ForwardRef(Button))
            classes={
              Object {
                "root": undefined,
              }
            }
            color="primary"
            data-testid="socialbutton-dataTestid-2"
            id="socialbutton-2"
            onClick={[Function]}
            variant="contained"
          >
            <FacebookIcon />
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        id="buttonSocial-3"
        item={true}
        key="buttonSocial-3"
        xs="auto"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionProps={
            Object {
              "timeout": 300,
            }
          }
          id="auth-button-3"
          placement="bottom"
          title="Visit the Library on Instagram"
        >
          <WithStyles(ForwardRef(Button))
            classes={
              Object {
                "root": undefined,
              }
            }
            color="primary"
            data-testid="socialbutton-dataTestid-3"
            id="socialbutton-3"
            onClick={[Function]}
            variant="contained"
          >
            <InstagramIcon />
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        id="buttonSocial-4"
        item={true}
        key="buttonSocial-4"
        xs="auto"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionProps={
            Object {
              "timeout": 300,
            }
          }
          id="auth-button-4"
          placement="bottom"
          title="Visit the Library on YouTube"
        >
          <WithStyles(ForwardRef(Button))
            classes={
              Object {
                "root": undefined,
              }
            }
            color="primary"
            data-testid="socialbutton-dataTestid-4"
            id="socialbutton-4"
            onClick={[Function]}
            variant="contained"
          >
            <YouTubeIcon />
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))>
      <span
        key="internallabel-0"
      >
        <a
          data-testid="connect-feedback-link"
          href="https://support.my.uq.edu.au/app/library/feedback"
        >
          Feedback
        </a>
        <span>
           |  
        </span>
      </span>
      <span
        key="internallabel-1"
      >
        <a
          data-testid="connect-participate-link"
          href="https://web.library.uq.edu.au/about-us/participate-customer-research"
        >
          Help us improve
        </a>
        <span>
           |  
        </span>
      </span>
      <span
        key="internallabel-2"
      >
        <a
          data-testid="connect-sitemap-link"
          href="https://web.library.uq.edu.au/sitemap"
        >
          Site Map
        </a>
      </span>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={4}
    xs={12}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="givingLinks-0"
        md={8}
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          data-testid="connect-friends-link"
          fullWidth={true}
          key="givingLinks-0"
          onClick={[Function]}
          variant="contained"
        >
          Join Friends of the Library
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="givingLinks-1"
        md={8}
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          data-testid="connect-give-link"
          fullWidth={true}
          key="givingLinks-1"
          onClick={[Function]}
          variant="contained"
        >
          Give to the Library
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
